src/TopPage/DataExtension.php

```diff
@@ -205,13 +231,47 @@ public function withTopPageUpdate(bool $update, callable $callback)
         }
     }
 
+    /**
+     * Use this to wrap any code which is supposed to run with fixed top page
+     * Useful when top page is known upfront and doesn't need to be determined
+     * For example: model duplication where parent is assigned and saved only after the duplication is done
+     * It's not possible to determine top page in such case however it might be possible to know the top page
+     * even before the operation starts from the specific context
+     * Setting the page id to 0 disables this feature
+     *
+     * @param int $topPageId
+     * @param callable $callback
+     * @return mixed
+     */
+    public function withFixedTopPage(int $topPageId, callable $callback)
+    {
+        $original = $this->fixedTopPageID;
+        $this->fixedTopPageID = $topPageId;
+
+        try {
+            return $callback();
+        } finally {
+            $this->fixedTopPageID = $original;
+        }
+    }
+
+    /**
+     * Get the ID of a page which is currently set as the fixed top page
+     *
+     * @return int
+     */
+    protected function getFixedTopPageId(): int
```

Comment:
Convention is to use `ID` rather that `Id`

```suggestion
    protected function getFixedTopPageID(): int
```